CRM/Civicase/Hook/PostProcess/AttachEmailActivityToAllCases.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * Add bulk email as an activity to all the selected cases.
+ */
+class CRM_Civicase_Hook_PostProcess_AttachEmailActivityToAllCases {
+
+  /**
+   * Add bulk email as an activity to all the selected cases.
+   *
+   * @param string $formName
+   *   The class name of the submitted form.
+   * @param object $form
+   *   The submitted form instance.
+   */
+  public function run($formName, $form) {
+    if (!$this->shouldRun($formName)) {
+      return;
+    }
+
+    $this->addActivityToAllSelectedCases();
+  }
+
+  /**
+   * Add bulk email as an activity to all the selected cases.
+   */
+  public function addActivityToAllSelectedCases() {
+    $firstCaseId = CRM_Utils_Array::value('caseid', $_GET, '0');
+    $allCaseIds = array_diff(explode(',', CRM_Utils_Array::value('allCaseIds', $_GET, '0')),
+      [$firstCaseId]);
+    try {
+      $activity = civicrm_api3('Activity', 'get', [
+        'case_id' => $firstCaseId,
+        'options' => [
+          'limit' => 1,
+          'sort' => "id DESC",
+        ],
+        'sequential' => 1,
+        'return' => ['id'],
+      ]);
+    }
+    catch (Throwable $ex) {
+      return;
+    }
+    $activityId = !empty($activity['id']) ? $activity['id'] : 0;
+    if ($activityId) {
+      foreach ($allCaseIds as $caseId) {
+        $caseActivity = new CRM_Case_DAO_CaseActivity();
+        $caseActivity->case_id = $caseId;
+        $caseActivity->activity_id = $activityId;
+        $caseActivity->save();
```

Comment:
Why are we not using the API for attaching the case Id to the activity here? 

```php
$result = civicrm_api3('Activity', 'create', [
  'id' => "sss",
  'case_id' => 2,
]);

```

Reply:
I tried that but that was not working for some reason

Reply:
This doesn't work for me too. It's strange and maybe a bug with core. Can you timebox a little time to check if you can find the reason why this doesn't work? Maybe we may need to raise this with core at some point.

Reply:
@shahrukh-compucorp , were you able to check this?